### PR TITLE
feat(share): add share extension template

### PR DESCRIPTION
**Pull request by gregorycat**
MP-456

### CODE REVIEW

**Review by germainruget — DISMISSED** [review]
Looks good !

As for now, like for the shareToSlack or shareToTeams, only the object of type Preview is pass as shareable content, which contain a title a link and an id like you did in the value props of the component.
Don't know if we will add an image and a description, if not it could be misleading to show those elements in the example.

**Review by LaurentMox — APPROVED** [review]
LGTM

File: template-share-extension/src/config.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/**
+ *  Create-LumApps-Widget config file
+ *  update the following to fit your needs
+ */
+
+export const useGlobalSettings = true;
```
Comment (LaurentMox) [inline template-share-extension/src/config.js:6]: Can we remove this please?

File: template-share-extension/src/config.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    //application,
+    links,
+    components,
+    isPublic,
+    public: isPublic,
+    whitelist,
+    category: 'share_to',
+};
+
+if (!useGlobalSettings) {
```
Comment (LaurentMox) [inline template-share-extension/src/config.js:87]: I would prefer if we can remove this condition. Partner can easily remove/add component to the list

File: template-share-extension/src/extension/Share.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            const lumapps = new Lumapps();
+            const { userLang: userLangPromise } = lumapps.context;
+
+            const userLang = await userLangPromise;
+            const isLangInTrad = Object.keys(messages).includes(userLang);
+
+            setLang(isLangInTrad ? userLang : 'en');
+        };
+        getContext();
+    }, [messages]);
```
Comment (LaurentMox) [inline template-share-extension/src/extension/Share.tsx:132]: I find it overkill. Can we just use the value from the hook ?
Something like we did in test-sdk extension:
```js
    const { displayLanguage } = useLanguage();
    const messages = useMemo(
        () => ({
            en: messagesEn,
            fr: messagesFr,
        }),
        [],
    );
    const lang = useMemo(() => (Object.keys(messages).includes(displayLanguage) ? displayLanguage : 'en'), [
        displayLanguage,
        messages,
    ]);
```

File: template-share-extension/src/extension/Share.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+    return (
+        <IntlProvider messages={messages[lang]} locale={lang}>
```
Comment (LaurentMox) [inline template-share-extension/src/extension/Share.tsx:135]: You can then update code like this:
`<IntlProvider messages={messages[lang as keyof typeof messages]} locale={lang}>`

File: template-share-extension/src/extension/ShareGlobalSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Render the widget Picsum settings form.
+ *
+ * @param {Object} props The settings component properties.
+ */
+export const ShareGlobalSettings = ({ properties = {}, exportProp }: any) => {
+    const [baseUrl, setBaseUrl] = useState(properties.baseUrl || defaultGlobalSettings.baseUrl);
+    const debouncedBaseUrl = useDebounce(baseUrl, 800);
+    useExportProps(debouncedBaseUrl, 'baseUrl', properties, exportProp);
```
Comment (LaurentMox) [inline template-share-extension/src/extension/ShareGlobalSettings.tsx:16]: It would be easier to read with some blank lines, as we are used to

**Review by LaurentMox — COMMENTED** [review]
Can we apply this small changes please?

**Review by LaurentMox — APPROVED** [review]
:love_letter:

**Pull request merged**